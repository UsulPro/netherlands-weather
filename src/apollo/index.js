import { ApolloClient } from 'apollo-client';
import fetch from 'node-fetch';
import { createHttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { window } from 'global';

const endpoint = process.env.ENDPOINT || 'https://api.graph.cool/simple/v1/cjpeioazq94e80143lvgo39w7';

export const client = () =>
  new ApolloClient({
    ssrMode: true,
    link: createHttpLink({
      uri: endpoint,
      fetch,
    }),
    cache: window
      ? new InMemoryCache().restore(window.__APOLLO_STATE__)
      : new InMemoryCache(),
  });
