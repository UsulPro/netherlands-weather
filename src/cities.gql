query getCities($city: String) {
  allCities(filter: { name: $city }) {
    name
  }
}
