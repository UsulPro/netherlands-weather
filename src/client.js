const React = require('react');
import ReactDOMServer from 'react-dom/server';

const App = ({ url }) => (
  <div>
    <h2>React App</h2>
    <span>on {url}</span>
    <hr/>
    <a href="some-url">test url</a>
  </div>
);


export const render = url => ReactDOMServer.renderToString(<App url={url} />);
