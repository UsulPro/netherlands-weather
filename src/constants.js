export const today = new Date('2014-08-08 00:00:00').toISOString();

export const theme = {
  headerBackground: '#336da7',
  panelBackground: '#6f3434',
};
