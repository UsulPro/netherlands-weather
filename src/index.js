import React from 'react';
import ReactDOM from 'react-dom';

import App from './app'

const renderApp = () => {
  console.log('â€‹App will start in 3s...');
  // TODO: It's just only for ssr testing, remove later
  setTimeout(
    () => ReactDOM.render(<App />, document.getElementById('app')),
    3000
  );
};

renderApp();
