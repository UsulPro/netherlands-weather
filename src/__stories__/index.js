import React from 'react';
import { storiesOf } from '@storybook/react';
import { ApolloProvider } from 'react-apollo';

import { client } from '../apollo/index';

import App from '../app';

const SApp = ({ url, client }) => (
  <ApolloProvider client={client}>
    <h2>SSR App</h2>
    <App url={url} />
  </ApolloProvider>
);

storiesOf('App', module).add('Home', () => (
  <SApp url={'/'} client={client()} />
));
