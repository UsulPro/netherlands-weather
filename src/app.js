import React from 'react';
import { ApolloProvider } from 'react-apollo';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

import { client } from './apollo/index';
import query from './cities.graphql'

const App = ({ url }) => (
  <div>
    <h2>React App</h2>
    <span>on {url}</span>
    <hr />
    <a href="amsterdam">amsterdam</a>
    <br />
    <a href="rotterdam">rotterdam</a>
    <hr />
    <Query
      // query={gql`
      //   query getCities($city: String) {
      //     allCities(filter: { name: $city }) {
      //       name
      //     }
      //   }
      // `}
      query={query}
      variables={{city: url.replace('/','')}}
    >
      {({ data, loading }) => <div>{JSON.stringify(data)}</div>}
    </Query>
  </div>
);

export default App;
