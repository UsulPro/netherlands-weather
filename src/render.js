import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { ApolloProvider, renderToStringWithData } from 'react-apollo';

import App from './app';
import { client } from './apollo/index';

// const SsrApp = (url) => (
//   <ApolloProvider client={client()}>
//     <h2>SSR App</h2>
//     <App url={url} />
//   </ApolloProvider>
// );

const SApp = ({ url, client }) => (
  <ApolloProvider client={client}>
    <h2>SSR App</h2>
    <App url={url} />
  </ApolloProvider>
);

const SsrApp = (url, client) => <SApp url={url} client={client} />;

const Sapp = url => <App url={url} />;

export const render = url =>
  renderToStringWithData(SsrApp(url, client())).then(content => {
    console.log('#### ​content ---->', content);
    const initialState = client().extract();
    console.log('​initialState', initialState);
    return content;
  });

// e xport const render = url =>
//   ReactDOMServer.renderToString(
//     <ApolloProvider client={client()}>
//       <App url={url} />
//     </ApolloProvider>
//   );
