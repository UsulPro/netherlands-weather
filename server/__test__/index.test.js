import { htmlTemplate } from '..';
import { render } from '../../src/render';

process.env.ENDPOINT =
  'https://api.graph.cool/simple/v1/cjpeioazq94e80143lvgo39w7';

it('SSR home page', async () => {
  expect(await htmlTemplate('/', render)).toMatchSnapshot();
});
it('SSR city page', async () => {
  expect(await htmlTemplate('/cities/amsterdam', render)).toMatchSnapshot();
});
