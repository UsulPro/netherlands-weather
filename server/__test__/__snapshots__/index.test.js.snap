// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR city page 1`] = `
"
      <!DOCTYPE html>
      <html lang=\\"en\\">
        <head>
          <meta charset=\\"utf-8\\" />
          <title>Weather</title>
        </head>
        <body>
          <div id=\\"app\\"><main><div style=\\"outline:none\\" tabindex=\\"-1\\" role=\\"group\\"><h2>AMSTERDAM</h2><br/><a href=\\"/\\">back</a><br/><br/><div>{&quot;allCities&quot;:[{&quot;name&quot;:&quot;Amsterdam&quot;,&quot;__typename&quot;:&quot;City&quot;}]}</div></div></main></div>
          <script>
            window.__APOLLO_STATE__ = {\\"ROOT_QUERY.allCities({\\\\\\"filter\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"amsterdam\\\\\\"}}).0\\":{\\"name\\":\\"Amsterdam\\",\\"__typename\\":\\"City\\"},\\"ROOT_QUERY\\":{\\"allCities({\\\\\\"filter\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"amsterdam\\\\\\"}})\\":[{\\"type\\":\\"id\\",\\"generated\\":true,\\"id\\":\\"ROOT_QUERY.allCities({\\\\\\"filter\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"amsterdam\\\\\\"}}).0\\",\\"typename\\":\\"City\\"}]}};
          </script>
          <script src=\\"/client.bundle.js\\"></script>
          <noscript>
            Without JavaScript you can't interact with this App.
          </noscript>
        </body>
      </html>
    "
`;

exports[`SSR home page 1`] = `
"
      <!DOCTYPE html>
      <html lang=\\"en\\">
        <head>
          <meta charset=\\"utf-8\\" />
          <title>Weather</title>
        </head>
        <body>
          <div id=\\"app\\"><main><div style=\\"outline:none\\" tabindex=\\"-1\\" role=\\"group\\"><h2>React App</h2><hr/><a href=\\"/cities/amsterdam\\">amsterdam</a><br/><a href=\\"/cities/rotterdam\\">rotterdam</a><hr/></div></main></div>
          <script>
            window.__APOLLO_STATE__ = {};
          </script>
          <script src=\\"/client.bundle.js\\"></script>
          <noscript>
            Without JavaScript you can't interact with this App.
          </noscript>
        </body>
      </html>
    "
`;
